x-pack/plugins/streams_app/public/components/streams_table/index.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBasicTable,
+  EuiBasicTableColumn,
+  EuiFlexGroup,
+  EuiIcon,
+  EuiLink,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import type { AbortableAsyncState } from '@kbn/observability-utils-browser/hooks/use_abortable_async';
+import { StreamDefinition } from '@kbn/streams-plugin/common';
+import React, { useMemo } from 'react';
+import { useStreamsAppRouter } from '../../hooks/use_streams_app_router';
+
+export function StreamsTable({
+  listFetch,
+  query,
+}: {
+  listFetch: AbortableAsyncState<{ definitions: StreamDefinition[] }>;
+  query: string;
+}) {
+  const router = useStreamsAppRouter();
+
+  const items = useMemo(() => {
+    return listFetch.value?.definitions ?? [];
+  }, [listFetch.value?.definitions]);
+
+  const filteredItems = useMemo(() => {
+    if (!query) {
+      return items;
+    }
+
+    return items.filter((item) => item.id.toLowerCase().startsWith(query.toLowerCase()));
```

Comment:
should we only match on the prefix? I expected this to match substrings in general.